src/main/java/com/openlattice/chronicle/services/enrollment/EnrollmentManagerImpl.java

```diff
@@ -0,0 +1,389 @@
+                            .getOrDefault( STATUS_FQN, Set.of( ParticipationStatus.UNKNOWN.toString() ) ).iterator()
+                            .next().toString() )
+                    .map( ParticipationStatus::valueOf )
+                    .findFirst().orElse( ParticipationStatus.UNKNOWN );
+
+        } catch ( Exception e ) {
+            logger.error( "unable to get participation status for participant: {}, study: {}, organization: {}.",
+                    participantId,
+                    studyId,
+                    organizationId, e );
+            return ParticipationStatus.UNKNOWN;
+        }
+    }
+
+    @Override
+    public boolean isNotificationsEnabled( UUID organizationId, UUID studyId ) {
+        logger.info( "Checking notifications enabled on studyId = {}, organization = {}", studyId, organizationId );
+
+        try {
+            ApiClient apiClient = apiCacheManager.prodApiClientCache.get( ApiClient.class );
+            SearchApi searchApi = apiClient.getSearchApi();
+
+            // get entity set ids
+            UUID studyESID = commonTasksManager.getEntitySetId( organizationId, CHRONICLE, STUDIES, STUDY_ES );
+            UUID notificationESID = commonTasksManager
+                    .getEntitySetId( organizationId, CHRONICLE, NOTIFICATION, NOTIFICATION_ES );
+            UUID partOfESID = commonTasksManager.getEntitySetId( organizationId, CHRONICLE, PART_OF, PART_OF_ES );
+
+            // ensure study exists
+            UUID studyEKID = Preconditions
+                    .checkNotNull( commonTasksManager.getStudyEntityKeyId( organizationId, studyId ),
+                            "study entity must exist" );
+
+            Map<UUID, List<NeighborEntityDetails>> neighbors = searchApi
+                    .executeFilteredEntityNeighborSearch(
+                            studyESID,
+                            new EntityNeighborsFilter(
+                                    ImmutableSet.of( studyEKID ),
+                                    java.util.Optional.of( ImmutableSet.of( notificationESID ) ),
+                                    java.util.Optional.of( ImmutableSet.of( studyESID ) ),
+                                    java.util.Optional.of( ImmutableSet.of( partOfESID ) )
+                            )
+                    );
+
+            /*
+             * studies with notifications enabled have 'ol.id' property in the corresponding
+             * associationDetails set to the value of the studyId
+             * For each study, there is can only be at most 1 notification -> partof -> study association,
+             * therefore it suffices to explore the first neighbor
+             */
+            return neighbors
+                    .values()
+                    .stream()
+                    .flatMap( Collection::stream )
+                    .anyMatch( neighbor -> neighbor.getAssociationDetails().getOrDefault( OL_ID_FQN, Set.of( "" ) )
```

Comment:
this is gonna be constructing a singleton set on each iteration through each neighbor. Perf/memory overhead would be improved by making this a private static constant up top, or (cleaner) making this function return false if neighbor.getAssociationDetails() does not contain OL_ID_FQN

Reply:
that `getFirstEntityValue` function would be handy here :)